File: backend/app/api/routes/certification.py
"""HTTP endpoint for public certifications."""

from typing import Annotated

from fastapi import APIRouter, Depends

from app.api.dependencies.certification import get_certification_service
from app.api.schemas.certification import CertificationResponse
from app.application.services.certification_service import CertificationService

router = APIRouter(prefix="/certifications", tags=["certifications"])


@router.get("", response_model=list[CertificationResponse])
async def get_certifications(
    service: Annotated[CertificationService, Depends(get_certification_service)],
) -> list[CertificationResponse]:
    certifications = await service.get_public_certifications()
    return [CertificationResponse.model_validate(item) for item in certifications]

File: frontend/src/lib/api/services/certification.ts
import { apiClient } from "@/lib/api/client";
import { certificationsSchema } from "@/schemas/certification";
import type { Certification } from "@/types/certification";

export async function getCertifications(): Promise<Certification[]> {
  return certificationsSchema.parse(await apiClient.get("/api/v1/certifications"));
}
